### DISCUSSION TITLE
Can we create custom roles and generate a Barer tokens with them and use the tokens in edge function to access the supabase database?

### DISCUSSION THREAD

**Discussion by deepak-coding-art** [supabase/supabase#13486] [Questions] [answered]
Hi,
I am new so need some help with this.
I want to create a new role and create Barer token with that role and use the role in RLS as `auth.role() = "stripe-bot"` . Use this Barer token in edge function when `createClient` to access the tables with this role.

**Comment by GaryAustin1** [comment, marked as answer]
Related https://github.com/orgs/supabase/discussions/11948

**Reply by deepak-coding-art** [reply]
Thanks for help
